Source/WebCore/rendering/RenderBoxModelObject.h

```diff
@@ -226,7 +226,8 @@ class RenderBoxModelObject : public RenderLayerModelObject {
     bool borderObscuresBackgroundEdge(const FloatSize& contextScale) const;
     bool borderObscuresBackground() const;
 
-    bool hasAutoHeightOrContainingBlockWithAutoHeight() const;
+    enum class RegisterPercentageDescendant : bool { No, Yes };
```

Comment:
Webkit already has UpdatePercentageHeightDescendants

Reply:
Sweet! Will update it.

Reply:
@alanbaradlay - thanks for your help, now it works. :-)